test/e2e/network/loadbalancer.go

```diff
@@ -1236,6 +1243,123 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
+		timeout := e2eservice.LoadBalancerLagTimeoutDefault
+		if framework.ProviderIs("aws") {
+			timeout = e2eservice.LoadBalancerLagTimeoutAWS
+		}
+		e2eservice.TestReachableHTTP(lbNameOrAddress, svcPort, timeout)
+
+		ginkgo.By("Starting a goroutine to continuously hit the DaemonSet's pods through the service's load balancer")
+		var totalRequests uint64 = 0
+		var failedRequests uint64 = 0
+		done := make(chan struct{})
+		defer close(done)
+		go func() {
+			defer ginkgo.GinkgoRecover()
+
+			wait.Until(func() {
+				totalRequests += 1
+				client := &http.Client{
+					Transport: utilnet.SetTransportDefaults(&http.Transport{
+						DisableKeepAlives: true,
+					}),
+					Timeout: 5 * time.Second,
+				}
+				ipPort := net.JoinHostPort(lbNameOrAddress, strconv.Itoa(svcPort))
+				msg := "hello"
+				url := fmt.Sprintf("http://%s/echo?msg=%s", ipPort, msg)
+				resp, err := client.Get(url)
+				if err != nil {
+					framework.Logf("Got error testing for reachability of %s: %v", url, err)
+					failedRequests += 1
+					return
+				}
+				defer resp.Body.Close()
+				if resp.StatusCode != http.StatusOK {
+					framework.Logf("Got bad status code: %d", resp.StatusCode)
+					failedRequests += 1
+					return
+				}
+				body, err := io.ReadAll(resp.Body)
+				if err != nil {
+					framework.Logf("Got error reading HTTP body: %v", err)
+					failedRequests += 1
+					return
+				}
+				if string(body) != msg {
+					framework.Logf("The response body does not contain expected string %s", string(body))
+					failedRequests += 1
+					return
+				}
+			}, time.Duration(0), done)
+		}()
+
+		ginkgo.By("Triggering DaemonSet rolling update several times")
+		// Time to complete the rolling update is proportional to the number of nodes in the cluster
+		retryTimeout := dsRetryTimeout + time.Duration(len(nodeNames)*30)*time.Second
+		for i := 1; i <= 5; i++ {
+			framework.Logf("Update daemon pods environment: [{\"name\":\"VERSION\",\"value\":\"%d\"}]", i)
+			patch := fmt.Sprintf(`{"spec":{"template":{"spec":{"containers":[{"name":"%s","env":[{"name":"VERSION","value":"%d"}]}]}}}}`, ds.Spec.Template.Spec.Containers[0].Name, i)
+			ds, err = cs.AppsV1().DaemonSets(ns).Patch(context.TODO(), name, types.StrategicMergePatchType, []byte(patch), metav1.PatchOptions{})
+			framework.ExpectNoError(err)
+
+			framework.Logf("Check that daemon pods are still running on every node of the cluster.")
+			err = wait.PollImmediate(dsRetryPeriod, retryTimeout, e2edaemonset.CheckDaemonPodOnNodes(f, ds, nodeNames))
+			framework.ExpectNoError(err, "error waiting for daemon pods to start")
+		}
+
+		// assert that the load balancer address is still reachable after the rolling updates are finished
+		e2eservice.TestReachableHTTP(lbNameOrAddress, svcPort, timeout)
+
+		// assert that the HTTP requests success rate is above the acceptable threshold
+		var successRate float64 = (float64(totalRequests) - float64(failedRequests)) / float64(totalRequests)
```

Comment:
this is going to race with the goroutine, syncLoadInt64() will solve it, same for increasing the counter

Reply:
Updated